addons/web/static/src/js/views/basic/basic_controller.js

```diff
@@ -331,6 +332,23 @@ var BasicController = AbstractController.extend(FieldManagerMixin, {
         var state = this.model.get(this.handle);
         return this.renderer.confirmChange(state, id, fields, e);
     },
+    /**
+     * Ask the user to confirm he wants to save the record
+     * @private
+     */
+    _confirmSaveNewRecord: function() {
+        var message = _t("You need to save this new record before editing the translation. Do you want to proceed?");
+        var def;
+        def = new Promise(function (resolve, reject) {
+            var dialog = Dialog.confirm(self, message, {
+                title: _t("Warning"),
+                confirm_callback: resolve.bind(self, true),
+                cancel_callback: reject,
+            });
+            dialog.on('closed', def, reject);
```

Comment:
The second argument is the component listening to the event, so:
`dialog.on('closed', self, def.reject.bind(def));`